hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ContainerReplicaPendingOps.java

```diff
@@ -48,10 +50,19 @@ public class ContainerReplicaPendingOps {
       pendingOps = new ConcurrentHashMap<>();
   private final Striped<ReadWriteLock> stripedLock = Striped.readWriteLock(64);
 
-  public ContainerReplicaPendingOps(final ConfigurationSource conf,
-      Clock clock) {
+  private final ExpiredContainerReplicaOpScrubber
+      expiredContainerReplicaOpScrubber;
+
+  public ContainerReplicaPendingOps(
+      final ConfigurationSource conf,
+      final Clock clock,
+      final SCMServiceManager serviceManager,
+      final SCMContext scmContext) {
```

Comment:
If we are going to have a separate "scrubber" class, then I don't think ContainerReplicaPendingOps needs a dependency on SCMServiceManager or SCMContext. It also doesn't need to keep a reference to the Scrubber thread.

The scrubber thread will be the "service" that needs registered with ServiceManager, and it will need a reference to the ContainerReplicaPendingOps instance. Even the scrubber does not need a dependency on ServiceManager - we can register it like:

```
Scrubber scrubber = new Scrubber(containerReplicaPendingOps);
serviceManager.register(scrubber);
```

That reduces the dependencies all the class have.

Reply:
thanks for the suggestion, looks good.  will do it